model/adjuster/span_merger.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adjuster
+
+import (
+	"github.com/jaegertracing/jaeger/model"
+)
+
+// MergeSpans returns an Adjuster that merges Jaeger spans with the same spanID.
+// Duplicate spans that have conflicting span.kind annotations are not merged.
+//
+// MergeSpans assumes that the duration field in a span is monotonically increasing for a span with the
+// same spanID. The span with the longest duration wins, and is selected in entirety.
+// TODO: Granular merging of spans
```

Comment:
what is this referring to?

Reply:
This refers to your suggestion on span merging rules: https://github.com/jaegertracing/jaeger/issues/729#issuecomment-370947209

The current implementation actually is not merging the spans, but just picking the span with the longest duration since the assumption is, that no deltas are sent, but always full spans.

This makes client implementations easier, since they don't need to care about any state, that may have been flushed already. Disadvantage is, that it needs more storage, because tags and logs will be duplicated for every intermediate span in the storage layer.

In general, granular merging only makes sense, if the client implementations are smart enough to send deltas in order to save network traffic and storage, but I am not sure, if this is not overcomplicating client logic, so I am not sure, if this will ever be needed.

Reply:
Ah, I didn't realize, that Spans don't have getters for their properties. So actually just flushing span data and only keeping spanId and traceId is actually not that hard